compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -8514,8 +8514,7 @@ private BType checkTupleIndexBasedAccess(BLangIndexBasedAccess accessExpr, BTupl
                 return checkTupleFieldType(tuple, getConstIndex(indexExpr).intValue());
             }
 
-            BTupleType tupleExpr = (BTupleType) accessExpr.expr.getBType();
-            LinkedHashSet<BType> tupleTypes = collectTupleFieldTypes(tupleExpr, new LinkedHashSet<>());
+            LinkedHashSet<BType> tupleTypes = collectTupleFieldTypes(tuple, new LinkedHashSet<>());
```

Comment:
Dont we need to collect rest type too?

Reply:
Let's create an issue for this error if this is not related.

```Ballerina
type IntTuple2 [int...];

const int index = 0;
public function main() {
    IntTuple2 intTuple = [1, 2, 3, 5];
    var x = intTuple[index + 3];
}
```
ERROR [t2.bal:(8:5,8:33)] unused variable 'x' with inferred type including error

Reply:
@MaryamZi  This is a bug right. But strangely I could not create a sample test case to make this a visible problem to write a decent code.

Reply:
Yeah, the type doesn't seem to be set properly.

```ballerina
public function main() {
    [int...] intTuple = [1, 2, 3, 5];
    int index = 0;
    var x = intTuple[index]; // unused variable error

    var y = intTuple[index];
    string _ = y; // no error
}
```

Reply:
I think we can fix this separately. Since the original issue seems to be when using member access on a type reference type for a tuple type.

@SasinduDilshara, can you also update the titles of the PR and the original issue (also adding a comment) to reflect the actual issues/fix?

Reply:
Updated

Reply:
Created a separate issue for this scenario. https://github.com/ballerina-platform/ballerina-lang/issues/36946

Reply:
Fixed 669f73c1135bfc854d66a95d1cff23aa80b6edec